### PR TITLE
Is first run

### PR DESCRIPTION
**Pull request by geof90**
Add support for isFirstRun.

@lostintangent

### CODE REVIEW

**Comment by lostintangent** [comment]
LGTM!

**Comment by lostintangent** [comment]
LGTM!

File: CodePush.m [hunk 1]
```diff
@@ -177,6 +179,18 @@ - (void)startRollbackTimer:(int)rollbackTimeout
...
 }
 
+RCT_EXPORT_METHOD(isFirstRun:(NSString *)packageHash
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    NSError *error;
+    BOOL isFirstRun = (nil != packageHash
+                  && [packageHash length] > 0
+                  && [packageHash isEqualToString:[CodePushPackage getCurrentPackageHash:&error]]
```
Comment (dtivel) [inline CodePush.m]: It looks like getCurrentPackageHash and getPreviousPackageHash would both null dereference if there is an error.

File: CodePush.m [hunk 2]
```diff
@@ -177,6 +179,18 @@ - (void)startRollbackTimer:(int)rollbackTimeout
...
+                  && didUpdate);
```
Comment (dtivel) [inline CodePush.m]: Make "didUpdate" the first expression to be evaluated.  It's a simple Boolean check that could make getting the current package hash unnecessary.

Remove the surrounding parentheses.

File: CodePush.m (hunk 1 above)
Comment (geof90) [inline CodePush.m]: Based on the developer guide, I think this is how methods that can return errors are handled?

https://developer.apple.com/library/ios/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/CreateCustomizeNSError/CreateCustomizeNSError.html#//apple_ref/doc/uid/TP40001806-CH204-BAJIIGCC

**Comment by dtivel** [comment]
looks good

**Pull request merged**